darts/tests/models/forecasting/test_classifier_model.py

```diff
@@ -0,0 +1,713 @@
+        )
+
+        assert model_instance.input_dim == {
+            "target": 1,
+            "past": 2,
+            "future": None,
+        }
+
+        with pytest.raises(ValueError):
+            prediction = model_instance.predict(n=len(test_y) + 2)
+
+        # while it should work with n = 1
+        prediction = model_instance.predict(n=1)
+        assert len(prediction) == 1
+
+    @pytest.mark.parametrize("clf_params", process_model_list(classifiers))
+    def test_labels_constraints(self, clf_params):
+        clf, kwargs = clf_params
+        model = clf(lags_past_covariates=2, **kwargs)
+
+        # Classification forecasting models do not accept continuous labels
+        with pytest.raises(ValueError):
+            model.fit(
+                series=self.sine_univariate1, past_covariates=self.sine_univariate1
+            )
+
+        # XGBClassifierModel require labels to be integers between 0 and n_classes
+        if type(clf) is XGBClassifierModel:
+            # negative labels
+            with pytest.raises(ValueError):
+                model.fit(
+                    series=self.sine_univariate1_cat - 5,
+                    past_covariates=self.sine_univariate1,
+                )
+
+            # labels not between 0 and n_classes
+            with pytest.raises(ValueError):
+                model.fit(
+                    series=self.sine_univariate1_cat + 1,
+                    past_covariates=self.sine_univariate1,
+                )
+
+        # Single label
+        if type(clf) in [
+            SVC,
+            GaussianProcessClassifier,
+            QuadraticDiscriminantAnalysis,
+        ] or type(model.model) in [LogisticRegression]:
+            with pytest.raises(ValueError):
+                model.fit(
+                    series=self.sine_univariate1_cat - self.sine_univariate1_cat,
+                    past_covariates=self.sine_univariate1,
+                )
+
+    @pytest.mark.parametrize("clf_params", process_model_list(classifiers))
+    def test_warning_raised_on_lags_and_target_not_cat(self, clf_params, caplog):
+        clf, kwargs = clf_params
+        with caplog.at_level(logging.WARNING):
+            model = clf(lags=2, **kwargs)
+            if isinstance(model, SKLearnModelWithCategoricalFeatures):
+                assert not any(
+                    record.levelname == "WARNING" for record in caplog.records
+                )
+            else:
+                assert any(record.levelname == "WARNING" for record in caplog.records)
+                assert any([
+                    message.startswith(
+                        "This model will treat the target `series` data/label as a numerical feature"
+                    )
+                    for message in caplog.messages
+                ])
+
+    @pytest.mark.parametrize("clf_params", process_model_list(classifiers))
+    def test_categorical_target_passed_to_fit_correctly(self, clf_params):
```

Comment:
Nice :) Could we try only running this for lgbm and catboost? Otherwise it will run a test for each model even though nothing is checked